zeppelin-server/src/test/java/org/apache/zeppelin/rest/AbstractTestRestApi.java

```diff
@@ -171,10 +171,16 @@ protected static void startUp() throws Exception {
 
         String sparkHome = getSparkHome();
         if (sparkHome != null) {
-          sparkIntpSetting.getProperties().setProperty("master", "spark://" + getHostname() + ":7071");
+          if (System.getenv("SPARK_MASTER") != null) {
+            sparkIntpSetting.getProperties().setProperty("master", System.getenv("SPARK_MASTER"));
+          } else {
+            sparkIntpSetting.getProperties()
+                    .setProperty("master", "spark://" + getHostname() + ":7071");
+          }
```

Comment:
It is for local system test when user want to run it in other modes (e.g. yarn-client).